provisioning/image_customization.go

```diff
@@ -0,0 +1,164 @@
+				Name:  name,
+				Value: "http://" + hostIP,
+			}
+		}
+		return corev1.EnvVar{
+			Name: name,
+		}
+	}
+}
+
+func createImageCustomizationContainer(images *Images, info *ProvisioningInfo) corev1.Container {
+	container := corev1.Container{
+		Name:            "image-customization-controller",
+		Image:           images.ImageCustomizationController,
+		Command:         []string{"/image-customization-controller"},
+		ImagePullPolicy: "IfNotPresent",
+		Env: []corev1.EnvVar{
+			setIronicBaseUrl(ironicBaseUrl, info),
+			{
+				Name:  ironicAgentImage,
+				Value: images.IronicAgent,
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("5m"),
+				corev1.ResourceMemory: resource.MustParse("50Mi"),
+			},
+		},
+	}
+	return container
+}
+
+func newImageCustomizationPodTemplateSpec(info *ProvisioningInfo, labels *map[string]string) *corev1.PodTemplateSpec {
+	containers := []corev1.Container{
+		createImageCustomizationContainer(info.Images, info),
+	}
+
+	tolerations := []corev1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/master",
+			Effect:   corev1.TaintEffectNoSchedule,
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:      "CriticalAddonsOnly",
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:               "node.kubernetes.io/not-ready",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+		{
+			Key:               "node.kubernetes.io/unreachable",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+	}
+
+	return &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: podTemplateAnnotations,
+			Labels:      *labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers:        containers,
+			HostNetwork:       true,
+			DNSPolicy:         corev1.DNSClusterFirstWithHostNet,
+			PriorityClassName: "system-node-critical",
+			NodeSelector:      map[string]string{"node-role.kubernetes.io/master": ""},
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: pointer.BoolPtr(false),
+			},
+			ServiceAccountName: "cluster-baremetal-operator",
+			Tolerations:        tolerations,
+		},
+	}
+}
+
+func newImageCustomizationDeployment(info *ProvisioningInfo) *appsv1.Deployment {
+	selector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"k8s-app":    metal3AppName,
+			cboLabelName: stateService,
+		},
+	}
+	podSpecLabels := map[string]string{
+		"k8s-app":    metal3AppName,
+		cboLabelName: stateService,
+	}
+	template := newImageCustomizationPodTemplateSpec(info, &podSpecLabels)
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "metal3ImageCustomization",
+			Namespace: info.Namespace,
+			Annotations: map[string]string{
+				cboOwnedAnnotation: "",
```

Comment:
I doubt this annotation is still needed

Reply:
Yeah, this was for the handshake with the MAO. Removed it.